### PR TITLE
CTPI-240 : Add retry strategy to delete product type due to Concurren…

### PR DESCRIPTION
**Pull request by leungkinghin**
Implement retry strategy for product type deletion and update for integration test

### CODE REVIEW

File: src/integration-test/java/com/commercetools/project/sync/util/ProductTypeITUtils.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import io.sphere.sdk.producttypes.ProductType;
+import io.sphere.sdk.producttypes.commands.ProductTypeDeleteCommand;
+import io.sphere.sdk.producttypes.queries.ProductTypeQuery;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+import javax.annotation.Nonnull;
+
+public final class ProductTypeITUtils {
```
[the remark this answers is not in the document]
Comment (leungkinghin) [reply src/integration-test/java/com/commercetools/project/sync/util/ProductTypeITUtils.java]: I removed ProductTypeITUtils. All private methods have been moved to IntegrationITUtils.java

[the remark this answers is not in the document]
Comment (ahmetoz) [reply src/integration-test/java/com/commercetools/project/sync/util/ProductTypeITUtils.java]: Commit: https://github.com/commercetools/commercetools-project-sync/pull/108/commits/be971373134b1743a1081b779454774c11d3e141

File: src/integration-test/java/com/commercetools/project/sync/util/IntegrationTestUtils.java [hunk 2]
```diff
@@ -134,7 +133,7 @@ private static void deleteProjectData(@Nonnull final SphereClient client) {
 
   private static void deleteProductTypes(@Nonnull final SphereClient ctpClient) {
     deleteProductTypeAttributes(ctpClient);
-    queryAndExecute(ctpClient, ProductTypeQuery.of(), ProductTypeDeleteCommand::of);
+    ProductTypeITUtils.deleteProductTypes(ctpClient);
   }
 
   private static void deleteProductTypeAttributes(@Nonnull final SphereClient ctpClient) {
```
[the remark this answers is not in the document]
Comment (ahmetoz) [reply src/integration-test/java/com/commercetools/project/sync/util/IntegrationTestUtils.java:178]: Commit: https://github.com/commercetools/commercetools-project-sync/pull/108/commits/70223cf9cd8815c05acfe8137d90e202649333a5

[the remark this answers is not in the document]
Comment (leungkinghin) [reply src/integration-test/java/com/commercetools/project/sync/util/IntegrationTestUtils.java:178]: https://github.com/commercetools/commercetools-project-sync/pull/108/commits/70223cf9cd8815c05acfe8137d90e202649333a5

**Pull request merged**